staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -91,8 +158,8 @@ var _ http.Hijacker = &fancyResponseWriterDelegator{}
 // 2. the response line containing:
 //    - the unique id from 1
 //    - response code
-func WithAudit(handler http.Handler, requestContextMapper request.RequestContextMapper, out io.Writer) http.Handler {
-	if out == nil {
+func WithAudit(handler http.Handler, requestContextMapper request.RequestContextMapper, backend AuditBackend, longRunningCheck filters.LongRunningRequestCheck) http.Handler {
```

Comment:
The meta-level logic isn't that complicated control-flow wise. I would prefer to have one impl and get that properly tested. The default policy will not be request or response level, so we have a natural feature toggle there.